File: simple-stack-example-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/Key.java
package com.zhuinden.simplestackdemoexamplefragments.application;

import android.os.Parcelable;
import android.support.annotation.DrawableRes;
import android.support.v4.app.Fragment;
import android.view.View;

/**
 * Created by Owner on 2017. 01. 12..
 */

public interface Key
        extends Parcelable {
    int layout();

    Fragment newFragment();

    String getFragmentTag();

    int menu();

    int navigationViewId();

    boolean isFabVisible();

    boolean shouldShowUp();

    View.OnClickListener fabClickListener(Fragment fragment);

    @DrawableRes
    int fabDrawableIcon();
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/Key.kt
package com.zhuinden.simplestackdemoexamplefragments.application

import android.os.Parcelable
import android.support.annotation.DrawableRes
import android.support.v4.app.Fragment
import android.view.View

/**
 * Created by Owner on 2017. 01. 12..
 */

interface Key : Parcelable {
    val fragmentTag: String

    val isFabVisible: Boolean

    fun layout(): Int

    fun newFragment(): Fragment

    fun menu(): Int

    fun navigationViewId(): Int

    fun shouldShowUp(): Boolean

    fun fabClickListener(fragment: Fragment): View.OnClickListener

    @DrawableRes
    fun fabDrawableIcon(): Int
}
